### ISSUE TITLE
Action groups must be fully qualified

### ISSUE THREAD

**Issue by mwhicks1** [cedar-policy/cedar#150]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/cedar-policy/cedar/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/cedar-policy/cedar/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### Bug Category

Schemas and Validation

### Describe the bug

An action's group membership requires a fully qualified name even when the specified group is within the same namespace.

### Expected behavior

When a referenced entity is defined in the same namespace in which it is referenced, I'd expect it not to require namespace qualification.

### Reproduction steps

Use the Cedar CLI to validate the policy `permit(principal,action,resource);` using the schema given below.

### Code Snippet

If you attempt to validate a policy using the following schema, you will see the error.
```
{
  "ExampleCo::Personnel": {
    "entityTypes": {},
    "actions": {
      "read": {},
      "viewPhoto": {
          "memberOf": [ 
              {
                "id": "read",
                "type": "Action"
             }
          ]
      }
    }
  }
}
```
The error goes away if you change the value of `type` to be `ExampleCo::Personnel::Action` instead.

### Log output

```
failed to parse schema from file dumb.schema: Undeclared actions: {"Action::\"read\""}
```

**Comment by john-h-kastner-aws** [comment]
As a workaround, the `type` field is not required here, and, when it is not present, the action takes the default namespace.  You can write `{"id": "read"}` and it will be interpreted as `{"type": "ExampleCo::Personnel::Action", "id": "read"}`.

We should still update the behavior of an unqualified `"type"` to match how to behaves in an `memberOfTypes` list and elsewhere.

**Comment by mwhicks1** [comment]
> As a workaround, the type field is not required here

Ack. Will update the docs to say this.

**Comment by john-h-kastner-aws** [comment]
We should be careful releasing this change since it's a breaking change for anyone relying on the current behavior: 

This schema would become an error since `NS::Action::"group"` isn't defined, or, if it is defined, a policy with scope constraint `action in NS::Action::"group"` would start to apply to `NS::Action::"act"` where it did not before, potentially causing validation errors 

```json
{
  "": {
    "entityTypes": { },
    "actions": {
      "group": {
        "appliesTo": {"principalTypes": "User"},
      }
    }
  },
  "NS": {
    "entityTypes": { },
    "actions": {
      "act": {
        "memberOf": [ {"type": "Action", "id": "group"} ]
      }
    }
  }
}
```

**Comment by mwhicks1** [comment]
I don't think this is a reason not to do it. The same potential confusion exists now with any entity, i.e.,
```
{
  "": {
    "entityTypes": { 
      "User": { ... }
   }
  },
  "NS": {
    "entityTypes": {
      "User": { ... },
      "Doc": {
        "shape": {
          "type": "Record",
          "attributes": {
            "owner" : {
              "type": "Entity",
              "name": "User" // which "User" is it?
            }
          }
        }
      }
    }
  }
}
```
We are saying that behavior should be consistent.

**Comment by john-h-kastner-aws** [comment]
Sure, but this is a change in behavior. We could call this a bug fix because it is inconsistent, but we should also consider if it's better to hold it for the next major version.

**Comment by aaronjeline** [comment]
Fix PR: https://github.com/cedar-policy/cedar/pull/151

**Comment by anwarmamat** [comment]
Closed by PR#151
